### DISCUSSION TITLE
suppress warnings for unsupported notification messages

### DISCUSSION THREAD

**Discussion by habamax** [yegappan/lsp#500] [General]
Hi, do you know how I can silence warning messages for unsupported notifications?

Am using it with godot lsp via `netcat` as in:

```
    if executable('netcat')
        g:LspAddServer([{
            name: 'gdscript',
            filetype: ['gdscript'],
            path: 'netcat',
            args: ['127.0.0.1', '6008'],
        }])
    endif
```

All I need works (completion, jump to definition, etc), but upon connecting to godot lsp I get quite a few warnings:

![image](https://github.com/yegappan/lsp/assets/234774/6b732bb1-2169-4614-bdd2-7fe6e82ba8ae)

Would appreciate if there is a method to suppress it. Thank you!

**Comment by D029-W** [comment]
I was reading the source code, and the following piece of code is located under `autoload/lsp/handler.vim`
```viml
# process unsupported notification messages
def ProcessUnsupportedNotif(lspserver: dict<any>, reply: dict<any>)
  util.WarnMsg($'Unsupported notification message received from the LSP server ({lspserver.name}), message = {reply->string()}')
enddef
```
This is the code sending the warning message that you see.
You could try to overwrite this function in your `vimrc`.

Or, you could try to add the notification that you don't want to see to the following in `autoload/lsp/handler.vim`
I recommend overwriting it in `vimrc`.
```viml
# process notification messages from the LSP server
export def ProcessNotif(lspserver: dict<any>, reply: dict<any>): void
  ...
  # Explicitly ignored notification messages (many of them are specific to a
  # particular language server)
  var lsp_ignored_notif_handlers: list<string> =
    [
      '$/progress',
      '$/status/report',
      ...,
    ]
```

**Reply by habamax** [reply]
Thx for the suggestion!

I don't think there is an easy way to override autoloaded function though.

**Reply by habamax** [reply]
I can add all the notification to the original function, but then next upgrade will override it.

I wish this was exposed as a global var:
https://github.com/yegappan/lsp/blob/dfc001ab109cb16ef4379d3d74554158bbebdcc5/autoload/lsp/handlers.vim#L97-L99
